include/libdnf5-cli/output/transactioninfo.hpp

```diff
@@ -30,6 +30,7 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 namespace libdnf5::cli::output {
 
 void print_transaction_info(libdnf5::transaction::Transaction & transaction);
+std::string generate_user_info_str(uint32_t user_id);
```

Comment:
Is there any reason why you want to add the function into public API? Be honest if there is no good reason I would prefer to move it in cpp to anonymous namespace and remove the declaration from HPP.